src/thunderbots/software/geom/util.h

```diff
@@ -486,3 +511,54 @@ Point getPointsMean(const std::vector<Point> &points);
  * @return the variance of the list of points
  */
 double getPointsVariance(const std::vector<Point> &points);
+
+
+/**
+ * Function returns the segment defined by the segment between the intersections of two
+ * Rays on a segment
+ *
+ * @param ray1 (Starting point and direction)
+ * @param ray2 (Starting point and direction)
+ * @param segment (Segment to find segment of intersection upon)
+ * @return Segment, the segment defined by the space between two intersecting rays on the
+ * segment parameter std::nullopt, if both rays don't intersect the segment, and the
+ * segment is not enclosed between the rays Segment, if one ray intersects the segment,
+ * but one of the segment parameters extremes are enclosed within the two rays
+ */
+std::optional<Segment> getIntersectingSegment(Ray ray1, Ray ray2, Segment segment);
+
+/**
+ * Function calculates whether the segment parameter is enclosed between the ray
+ * parameters. This means the entirety of the segment lays between the rays
+ *
+ * @param segment : segment parameter to calculate if its definition lies between the rays
+ * @param ray1 : Starting point and direction
+ * @param ray2 : Starting point an direction
+ * @return Segment: Returns the segment parameter if it is enclosed between ray1 and ray2
+ *         Returns std::nullopt of the ray is not completely enclosed between the rays, or
+ * not at all
+ */
+std::optional<Segment> segmentEnclosedBetweenRays(Segment segment, Ray ray1, Ray ray2);
+
+/**
+ * Function merges overlapping parallel segments into one combined segment
+ *
+ * @param segment1 : first segment
+ * @param segment2 : second segment
+ * @return Segment: Returns the merged segment if segment1 & segment2 are parallel and
+ * overlapping Returns std::nullopt if the segments aren't parallel or overlapping
+ */
+std::optional<Segment> mergeOverlappingParallelSegments(Segment segment1,
+                                                        Segment segment2);
+
+/**
+ * Function calculates if the segment parameters are redundant, for example, if segment2
```

Comment:
This seems nearly identical to the `mergeOverlappingParallelSegments` function above. How else would segments be redundant?

Reply:
The idea behind this function is it checks if one of the segments exists entirely within the other. mergeOverlappingParallelSegments() uses this function as one of the potential cases. mergeOverlappingParallelSegments() will combine 2 entirely or partially overlapping segments into 1 segment that contains the same space

Reply:
Maybe rename this to something like `mergeFullyOverlappingSegments` or something more similar to the other functions then? Naming it with "redundant" kinda threw me off, and I think this is more descriptive.

Reply:
That's fair. Renamed as suggested.